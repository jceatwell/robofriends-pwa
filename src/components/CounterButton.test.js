import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

describe('CounterButton Component', () => {
    let wrapper;

    beforeEach( () => {
        const stubColor = 'red';
        wrapper = shallow(<CounterButton color={stubColor}/>)
    });

    it('should compare to previous snapshot when test run', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should correctly increment counter (find by ID)', () => {
        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({count: 1});
    });

    it('should correctly increment counter (find by type)', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.state()).toEqual({count: 1});
    });    
});
