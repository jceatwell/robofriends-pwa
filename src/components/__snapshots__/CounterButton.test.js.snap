// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component should compare to previous snapshot when test run 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
